File: scripts/apply-rls-fix.ts
import { createClient } from '@supabase/supabase-js';
import fs from 'fs';
import dotenv from 'dotenv';

dotenv.config({ path: '.env.local' });

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL!;
const supabaseServiceKey = process.env.SUPABASE_SERVICE_ROLE_KEY!;

async function applyMigration() {
  const client = createClient(supabaseUrl, supabaseServiceKey, {
    auth: { autoRefreshToken: false, persistSession: false }
  });

  console.log('🔧 Applying RLS fix migration...');

  const sql = fs.readFileSync('supabase/migrations/038_fix_scheduling_rls_app_metadata.sql', 'utf-8');

  // Split by semicolons and execute each statement
  const statements = sql
    .split(';')
    .map(s => s.trim())
    .filter(s => s && !s.startsWith('--'));

  for (const statement of statements) {
    console.log(`Executing: ${statement.substring(0, 60)}...`);
    const { error } = await client.rpc('exec_sql', { sql: statement });

    if (error) {
      console.error('❌ Error:', error);
      // Try direct execution
      const response = await fetch(`${supabaseUrl}/rest/v1/rpc/exec`, {
        method: 'POST',
        headers: {
          'apikey': supabaseServiceKey,
          'Authorization': `Bearer ${supabaseServiceKey}`,
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({ query: statement }),
      });

      if (!response.ok) {
        console.error('❌ Failed:', await response.text());
      }
    } else {
      console.log('✅ Success');
    }
  }

  console.log('✅ Migration applied');
}

applyMigration().catch(console.error);

File: apply_rls_fix.py
#!/usr/bin/env python3
import requests

SUPABASE_URL = "https://rtwigjwqufozqfwozpvo.supabase.co"
SERVICE_KEY = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InJ0d2lnandxdWZvenFmd296cHZvIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc1NDI1MDMwMCwiZXhwIjoyMDY5ODI2MzAwfQ.e4U3aDv5GDIFiPlY_JcveGwbAT9p-ahiW_0hhoOUoY0"

headers = {
    "apikey": SERVICE_KEY,
    "Authorization": f"Bearer {SERVICE_KEY}",
    "Content-Type": "application/json",
    "Prefer": "return=minimal"
}

# Read the migration SQL
with open('/Users/travisetzler/Documents/GitHub/jobeye/supabase/migrations/100_fix_users_extended_rls_recursion.sql', 'r') as f:
    sql = f.read()

print("Applying RLS fix migration...")
print("-" * 80)

# Split by semicolons and execute each statement
statements = [s.strip() for s in sql.split(';') if s.strip() and not s.strip().startswith('--')]

for i, statement in enumerate(statements, 1):
    if not statement or statement.startswith('COMMENT'):
        continue

    print(f"\nExecuting statement {i}/{len(statements)}:")
    print(statement[:100] + "..." if len(statement) > 100 else statement)

    try:
        # Using the admin API to execute SQL directly
        response = requests.post(
            f"{SUPABASE_URL}/rest/v1/rpc/__execute_sql",
            headers=headers,
            json={"sql": statement + ";"}
        )

        if response.status_code in [200, 201, 204]:
            print(f"✓ Success")
        else:
            print(f"✗ Failed: {response.status_code}")
            print(f"Response: {response.text}")
    except Exception as e:
        print(f"✗ Error: {e}")

print("\n" + "-" * 80)
print("Migration application complete!")
